### PR TITLE
Connect speech locale with route locale.

### PR DESCRIPTION
**Pull request by StephanPartzsch**
- Uses the `Locale` given in `RouteOptions` to create the correct `AVSpeechSynthesisVoice`.
- Extends the ui in the example with a button to specify the `Locale` for the `Route`.

### CODE REVIEW

**Review by boldtrn — CHANGES_REQUESTED** [review]
Thanks for looking into this. I think it makes a lot of sense to allow configuring the locale of the voice output. 

Would you mind adding a Changelog entry for your change so that implementers know what they need to do.

**Comment by boldtrn** [comment]
Given the merge of #23, could you resolve the conflicts, then we can go ahead and merge this PR.

File: MapboxNavigation/RouteVoiceController.swift [hunk 1]
```diff
@@ -180,17 +180,23 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
+        
         lastSpokenInstruction = instruction
-        speak(instruction)
+        speak(instruction, with: speechLocale)
     }
     
     /**
      Reads aloud the given instruction.
      
      - parameter instruction: The instruction to read aloud.
```
[the remark this answers is not in the document]
Comment (StephanPartzsch) [reply MapboxNavigation/RouteVoiceController.swift:197]: Good catch! 
Done

File: Examples/Swift/Base.lproj/Main.storyboard
Comment (boldtrn) [inline Examples/Swift/Base.lproj/Main.storyboard:1]: With the Maplibre 6.0 update, the storyboard needs to be changed from using `MBNavigationMapView` to `MLNavigationMapView`

**Review by boldtrn — CHANGES_REQUESTED** [review]
I just checked out your branch and tried to run it, it crashes due to the missing view in the storyboard, due to the update to Maplibre 6.

File: MapboxNavigation/RouteVoiceController.swift [hunk 2]
```diff
@@ -180,17 +180,24 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
         routeProgress = notification.userInfo![RouteControllerNotificationUserInfoKey.routeProgressKey] as? RouteProgress
         assert(routeProgress != nil, "routeProgress should not be nil.")
 
-        guard let instruction = routeProgress!.currentLegProgress.currentStepProgress.currentSpokenInstruction else { return }
+        guard
+            let instruction = routeProgress?.currentLegProgress.currentStepProgress.currentSpokenInstruction,
+            let speechLocale = routeProgress?.route.routeOptions.locale
+        else {
+            return
```
[the remark this answers is not in the document]
Comment (StephanPartzsch) [reply MapboxNavigation/RouteVoiceController.swift]: Done

**Pull request merged**